File: raindrops/raindrops.go
package raindrops

import "strconv"

func Convert(num int) string {
  if num % 3 != 0 && num % 5 != 0 && num % 7 != 0 {
    return strconv.Itoa(num)
  }

  raindrop := []byte("")
  if num % 3 == 0 {
    raindrop = append(raindrop, "Pling"...)
  }
  if num % 5 == 0 {
    raindrop = append(raindrop, "Plang"...)
  }
  if num % 7 == 0 {
    raindrop = append(raindrop, "Plong"...)
  }

  return string(raindrop)
}

File: haskell/raindrops/src/Raindrops.hs
module Raindrops (convert) where

convert :: Int -> String
convert n =
  if null res then show n else res
  where
    res = concatMap (sound n) [(3, "Pling"), (5, "Plang"), (7, "Plong")]
    sound x (a, b)
      | mod x a == 0 = b
      | otherwise    = ""

